### PR TITLE
Jkl 198 msg buy storage bug fix & simulation implementation

### PR DESCRIPTION
**Pull request by dahn510**
Fixes bug that prevents user from buying storage.
Duration value of the MsgBuyStorage test cases are changed to time.Duration format.

### CODE REVIEW

**Comment by BiPhan4** [comment]
I'm not the appropriate person to review this PR as I've not been working on the new payment system. I defer to Justin and Erin here.

**Comment by TheMarstonConnell** [comment]
No proto changes?

**Comment by dahn510** [comment]
> No proto changes?

Nope

File: x/storage/simulation/utils.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	data = append(data, hashName)
+
+	tree, err := merkle.New(data)
+	if err != nil {
+		panic(err)
+	}
+
+	// Build proof
+	h = sha256.New()
+	_, err = io.WriteString(h, fmt.Sprintf("%d%x", 0, f))
```
[the remark this answers is not in the document]
Comment (TheMarstonConnell) [reply x/storage/simulation/utils.go:57]: Looks like this isn't handled 😳

**Comment by nuggetnchill** [comment]
the bug fix is good to go, I'll wait till we sort out the fail simulation test before approving this PR

**Pull request merged**